pinot-core/src/main/java/com/linkedin/pinot/core/startree/v2/AggregationFunctionColumnPairBuffer.java

```diff
@@ -0,0 +1,191 @@
+/**
+ * Copyright (C) 2014-2018 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.startree.v2;
+
+import java.util.List;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import com.linkedin.pinot.common.data.FieldSpec;
+import com.linkedin.pinot.core.common.datatable.ObjectType;
+import com.linkedin.pinot.core.segment.memory.PinotDataBuffer;
+import com.linkedin.pinot.core.common.datatable.ObjectCustomSerDe;
+
+
+public class AggregationFunctionColumnPairBuffer {
+
+  int _totalBytes;
+  protected final Object[] _values;
+  private final List<AggregationFunctionColumnPair> _aggFunColumnPairs;
+
+  public AggregationFunctionColumnPairBuffer(Object[] values, List<AggregationFunctionColumnPair> aggFunColumnPairs) {
```

Comment:
Pass in a list of functions instead of the function-column pairs

Reply:
done